sql/database.go

```diff
@@ -356,12 +585,16 @@ func (db *sqliteDatabase) getTx(ctx context.Context, initstmt string) (*sqliteTx
 	return tx, nil
 }
 
-func (db *sqliteDatabase) withTx(ctx context.Context, initstmt string, exec func(Transaction) error) error {
+func (db *sqliteDatabase) withTx(ctx context.Context, initstmt string, exec func(Transaction) error) (err error) {
 	tx, err := db.getTx(ctx, initstmt)
 	if err != nil {
 		return err
 	}
-	defer tx.Release()
+	defer func() {
+		if rErr := tx.Release(); rErr != nil {
+			err = errors.Join(err, fmt.Errorf("release tx: %w", rErr))
+		}
+	}()
```

Comment:
I believe we can do this also on a best effort basis:

```suggestion
	defer tx.Release()
```

The returned error afaik is never handled, so just adding more errors to the original issue will just cause the logged error to be longer.

Alternatively if the error returned from `tx.Release()` has the potential to be more severe then errors returned from `exec` maybe it should overwrite it? For instance if `exec` returns `ErrObjectExists` or `ErrNotFound` and `tx.Release()` returns a real error then a check for `error.Is(err, ErrObjectExists)` or `error.Is(err, ErrNotFound)` should not swallow that (more severe) error.

Reply:
Fixed. But it is worth noting that there's (theoretical) a possibility that `tx.Release()` returns an error for a successfully completed transaction. Right now this is not possible, but if `tx.Release()` evolves so that this becomes possible, we might miss an error here.
So I removed `errors.Join` but made the code return the error returned by `tx.Release()` if `err` is `nil`.